compiler/ETA/CodeGen/Constr.hs

```diff
@@ -41,18 +41,24 @@ cgTopRhsCon dflags id dataCon args = (cgIdInfo, genCode)
         -- typeFt = obj (tyConClass dflags (dataConTyCon dataCon))
         genCode = do
           loads <- mapM getArgLoadCode . getNonVoids $ zip maybeFields args
-          deps <- getArgReferences . getNonVoids $ zip maybeFields args
-          defineField $ mkFieldDef [Public, Static] qClName closureType
-          let field = (mkFieldRef modClass qClName closureType)
-          addInitStep (
-              new dataFt
-           <> dup dataFt
-           <> fold loads
-           <> invokespecial (mkMethodRef dataClass "<init>" fields void)
-           <> putstatic field
-           , field
-           , deps
-           )
+          defineField $ mkFieldDef [Private, Static] qClName closureType
+          let field = mkFieldRef modClass qClName closureType
+              loadCodes =
+                [
+                  new dataFt
+                , dup dataFt
+                , fold loads
+                , invokespecial (mkMethodRef dataClass "<init>" fields void)
+                , putstatic field
+                ]
+          defineMethod . mkMethodDef' [Public, Static] qClName (mkMethodDesc [] (Just closureType)) $ fold
```

Comment:
Yeah, that should be the class name of the class that contains the static method. `mkMethodDef` appends some local variables initialisation code in front of the code you provide and in this case since there are no local variables or arguments, so it doesn't seem to make a difference. I'll file an issue on `codec-jvm` to not export `mkMethodDef'`. Can you plug in the value returned from `getModClass` into that argument for consistency?